### ISSUE TITLE
Plugin fields and assets do not work in composer-installed k3

### ISSUE THREAD

**Issue by omz13** [getkirby/kirby#1094]
When developing plugins for k3, everything seems to work... except for "fields" when the kirby3 site itself has been installed using composer. @bnomei has also experienced this (so not just my environment.)

When trying to use a field, the message `The field type "whatever" does not exist` is shown instead of the field, and the console is showing errors for index.css/js:
```
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (index.css, line 0)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (index.js, line 0)
```

I did a quick look at it seems that .../media/plugins/.index.css and .index.js are not generated - I had a quick look into `PluginAssets.php` but couldn't see anything obviously wrong.

From another time I was experimenting, and the contents of the `assets` folder from a plugin were similarly not working (being copied to /media/plugins/vendor/plugin - related to #937 

To help, I've created a simple field-only plugin [omz13/whatever](https://github.com/omz13/kirby3-whatever), and made this available via packagist.

**Comment by georgobermayr** [comment]
I'm having this issue as well. The issue is also with a devkit setup when using the plugin boilerplate (https://github.com/k-next/demo-plugin): `The field type "currency" does not exist`.

**Comment by bnomei** [comment]
my guess it that this fails:

https://github.com/k-next/kirby/blob/e31bb4d7fa907da832d08dcee80f92a1d0ec6adc/src/Cms/PluginAssets.php#L100

maybe because `F::exists` is not the same as `file_exits` it does call some other static functions
https://github.com/k-next/kirby/blob/master/src/Toolkit/F.php#L507

simple check would be to replace `F::exists` with `file_exists` and try again. but i do not have the time to do that now.

**Comment by bnomei** [comment]
working locally on mamp pro 4.4.1 with apache (2.2.34, port:8888) with php 7.2.1.

**Comment by georgobermayr** [comment]
> simple check would be to replace `F::exists` with `file_exists` and try again. but i do not have the time to do that now.

At least for me this doesn't change anything in the behaviour. 

My environment: MAMP 5.1 with PHP 7.28

**Comment by omz13** [comment]
I saw this [thread in slack](https://getkirby.slack.com/archives/CBN859614/p1540542176005200) and I tried something similar with the result that I can now get things to work (locally) when I run through the proposed `router.php` with `php -S localhost:8888 router.php` (including the thumbnail issue which also happened with me has also stopped).... but, when I run the same in a production environment (where the server routes php - actually php-fpm - requests through `index.php`), the problem persists.

I've tried what I can to resolve it, but its eluding me.

If it would help, I have a test repo I can put up (see `blancmange` that I sent directly to you).

**Comment by bastianallgeier** [comment]
Hmm, I just tried really hard to reproduce this, but it's working fine for me. I used @omz13's plugin in a simple composer based setup and everything is loaded as expected. Is this still an issue in one of the latest nightlies?

**Comment by omz13** [comment]
TL;DR narrowed down to: something weird happens with php7.2.x and nginx or apache or caddy, or with php7.x and caddy.

I am still getting some weirdness with the latest build (3.0.0-beta-6.16).

caddy + php-fpm (7.1) in a vagrant container: plugin fields don't work 👎 and weirdness

👀  This behaviour is now happening under apache and nginx if I switch from php 7.1.20 to php 7.2.8.

apache + php 7.2 + MAMP 5 Pro = plugin fields not working 👎  and weirdness

caddy + php-fpm 7.1 in a production server: plugin fields don't work 👎

nginx + php 7.1 MAMP 5 Pro = it all works 👍 

nginx + php 7.2 MAMP 5 Pro = weirdness 👎  plugin fields not working

apache + php 7.1 + MAMP 5 Pro = it all works 👍 

nginx + php 7.1 in a production server = it all works 👍

**Comment by georgobermayr** [comment]
I'm no longer affected by this issue, because I switched to a Browserify based plugin setup close to the  ones developed by the community a while ago. So I didn't test this any further with the latest builds.

**Comment by bastianallgeier** [comment]
I'm pretty sure this was the same issue that we had when you copied plugins from one installation to the next and the cache didn't update. We fixed this in the meantime.
